storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -1577,63 +1577,76 @@ if GTID is enabled */
 bool
 write_current_binlog_file(MYSQL *connection)
 {
-	char *executed_gtid_set = NULL;
-	char *gtid_binlog_state = NULL;
 	char *log_bin_file = NULL;
 	char *log_bin_dir = NULL;
-	bool gtid_exists;
+	char *log_bin_index = NULL;
+	char *log_bin = NULL;
 	bool result = true;
 	char filepath[FN_REFLEN];
 
-	mysql_variable status[] = {
-		{"Executed_Gtid_Set", &executed_gtid_set},
-		{NULL, NULL}
-	};
+	std::string index_path;
+	std::string index_filename;
+	std::string binlog_path;
+	std::string binlog_dir;
+	size_t last_sep = 0;
 
 	mysql_variable status_after_flush[] = {
 		{"File", &log_bin_file},
 		{NULL, NULL}
 	};
 
 	mysql_variable vars[] = {
-		{"gtid_binlog_state", &gtid_binlog_state},
+		{"log_bin", &log_bin},
 		{"log_bin_basename", &log_bin_dir},
+        {"log_bin_index", &log_bin_index},
 		{NULL, NULL}
 	};
 
-	read_mysql_variables(connection, "SHOW MASTER STATUS", status, false);
 	read_mysql_variables(connection, "SHOW VARIABLES", vars, true);
 
-	gtid_exists = (executed_gtid_set && *executed_gtid_set)
-			|| (gtid_binlog_state && *gtid_binlog_state);
-
-	if (gtid_exists) {
+	if (strcmp(log_bin, "OFF") != 0) {
```

Comment:
PXB 8.0 gets the GTID information from p_s.log_status table. Where are we getting GTID information to be written to xtrabackup_binlog_pos file ?

Reply:
This is for the binlog files (write_current_binlog_file()), the binlog info is written in write_binlog_info() and uses a combination of "SHOW MASTER STATUS" and "SHOW VARIABLES".

Note that with this change we are backing up the binlog files all the time, not just when gtids are enabled.